frontend/src/components/new-car-card/new-car-card.tsx

```diff
@@ -1,35 +1,100 @@
-import { FC } from 'react';
+import React from 'react';
 
-import bmw from '@assets/images/bmw_logo.svg';
 import compare from '@assets/images/compare.svg';
-import heart from '@assets/images/heart.svg';
-import new_car_1 from '@assets/images/mock_car_picture.png';
+import { WishlistInput } from '@autoline/shared/common/types/types';
+import { ExtendedCarCardPropsType } from '@common/types/types';
+import { Heart } from '@components/common/icons/icons';
+import { clsx } from 'clsx';
 
 import styles from './styles.module.scss';
 
-const NewCarCard: FC = () => {
+const NewCarCard: React.FC<ExtendedCarCardPropsType> = (props) => {
+  const formatPriceNumber = (num: number): string => {
+    const price = num.toString().split('.');
+    let numberPart = price[0];
+    const floatPart = price.length > 1 ? '.' + price[1] : '';
+    const rgx = /(\d+)(\d{3})/;
+    while (rgx.test(numberPart)) {
+      numberPart = numberPart.replace(rgx, '$1' + ' ' + '$2');
+    }
+    return numberPart + floatPart;
+  };
+
+  const minPrices = props.car.pricesRanges.map(
+    (price: { price_start: number; price_end: number }) => price.price_start,
+  );
+  const minPrice = formatPriceNumber(Math.min.apply(null, minPrices));
+  const maxPrices = props.car.pricesRanges.map(
+    (price: { price_start: number; price_end: number }) => price.price_end,
+  );
+  const maxPrice = formatPriceNumber(Math.max.apply(null, maxPrices));
+
+  const handleCreate = (): void => {
+    const data: WishlistInput = {
+      modelId: props.type === 'model' ? props.car.id : undefined,
+      complectationId:
+        props.type === 'complectation' ? props.car.id : undefined,
+    };
+
+    props.createWishlist(data);
+  };
+
+  const handleDelete = (): void => {
+    props.deleteWishlist(props.car.id);
+  };
+
+  const handleLikeClick = (event: React.MouseEvent): void => {
+    event.stopPropagation();
+    props.isLiked ? handleDelete() : handleCreate();
+  };
+
+  let name = props.car.brand?.name.concat(' ', props.car.name);
+
+  if (props.car.complectationName) {
+    name = props.car.brand?.name.concat(' ', props.car.complectationName);
+  }
```

Comment:
Please use the template string here. Usage of the `contact` isn't recommended in the modern JS